src/ngraph/op/extractimagepatches.cpp

```diff
@@ -0,0 +1,157 @@
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates[0] > 0 && m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_padding == PadType::VALID || m_padding == PadType::SAME_LOWER ||
+            m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+
+    if (input_Pshape.is_dynamic())
+    {
+        set_input_is_relevant_to_shape(0);
+        auto output_Pshape = PartialShape::dynamic(4);
+        set_output_type(0, get_input_element_type(0), output_Pshape);
+    }
+    else
+    {
+        Shape input_shape = get_input_shape(0); // as input shape is static
+        size_t out_rows(0);
+        size_t out_cols(0);
+
+        if (m_patch_movement_strides[0] > input_shape[2] ||
+            m_patch_movement_strides[1] > input_shape[3])
+        {
+            if (m_padding == PadType::VALID)
+            {
+                out_rows = 0; // no padding allowed for VALID
+                out_cols = 0; // no padding allowed for VALID
+            }
+            else
+            {
+                out_rows = 1; // padding will get 1 patch
+                out_cols = 1; // padding will get 1 patch
+            }
+        }
+        else if (input_shape[2] == 0 || input_shape[3] == 0)
+        {
+            out_rows = 0;
+            out_cols = 0;
+        }
+        else if (m_padding == PadType::VALID)
+        {
+            out_rows =
+                ((input_shape[2] - (m_patch_selection_rates[0]) * (m_patch_sizes[0] - 1) - 1) /
+                 m_patch_movement_strides[0]) +
+                1;
+            out_cols =
+                ((input_shape[3] - (m_patch_selection_rates[1]) * (m_patch_sizes[1] - 1) - 1) /
+                 m_patch_movement_strides[1]) +
+                1;
+        }
+        else
+        {
+            out_rows = 1 + ((input_shape[2] - 1) / m_patch_movement_strides[0]);
+            out_cols = 1 + ((input_shape[3] - 1) / m_patch_movement_strides[1]);
+        }
+        Shape output_shape;
+        output_shape.push_back(input_shape[0]);
+        output_shape.push_back(input_shape[1] * m_patch_sizes[0] *
+                               m_patch_sizes[1]); // size[1]*size[2]*depth
+        output_shape.push_back(out_rows);
+        output_shape.push_back(out_cols);
+
+        if (input_shape[2] == 0 || input_shape[3] == 0)
+        {
+            output_shape = input_shape;
+        }
+
+        set_output_type(0, get_input_element_type(0), output_shape);
+    }
+}
+
+bool op::v3::ExtractImagePatches::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("sizes", m_patch_sizes);
+    visitor.on_attribute("strides", m_patch_movement_strides);
+    visitor.on_attribute("rates", m_patch_selection_rates);
+    visitor.on_attribute("padding", m_padding);
```

Comment:
The attribute has been renamed to "auto_pad"

Reply:
Thanks. updated the name.